Sources/Tuvi.Core.Mail.Impl/Protocols/SMTP/SMTPMailService.cs

```diff
@@ -20,11 +20,15 @@ public override async Task<string> SendMessageAsync(Message message, Cancellatio
         {
             try
             {
-                using (var mimeMessage = message.ToMimeMessage())
-                {
-                    await SmtpClient.SendAsync(mimeMessage, cancellationToken).ConfigureAwait(false);
-                    return mimeMessage.MessageId;
-                }
+                return await SendAsync().ConfigureAwait(false);
+            }
+            catch (System.IO.IOException)
+            {
+                // after exception connection may be lost
+                await RestoreConnectionAsync(cancellationToken).ConfigureAwait(false);
+
+                // retry once
+                return await SendAsync().ConfigureAwait(false);
```

Comment:
Should we use try, catch?

Reply:
No, if an exception is thrown on the second attempt, let it propagate to the user.

Reply:
What if the exception is different? (e.g. MailKit.ServiceNotConnectedException, MailKit.ServiceNotAuthenticatedException)

Reply:
Anyway.

Reply:
Ok

Reply:
Ok. Fixed :)